indykite/resource_consent.go

```diff
@@ -0,0 +1,119 @@
+		CreateContext: configCreateContextFunc(resourceConsentBuild, readContext),
+		ReadContext:   readContext,
+		UpdateContext: configUpdateContextFunc(resourceConsentBuild, readContext),
+		DeleteContext: configDeleteContextFunc(),
+		Importer: &schema.ResourceImporter{
+			StateContext: basicStateImporter,
+		},
+
+		Timeouts: defaultTimeouts(),
+		Schema: map[string]*schema.Schema{
+			locationKey:    locationSchema(),
+			customerIDKey:  setComputed(customerIDSchema()),
+			appSpaceIDKey:  setComputed(appSpaceIDSchema()),
+			nameKey:        nameSchema(),
+			displayNameKey: displayNameSchema(),
+			descriptionKey: descriptionSchema(),
+			createTimeKey:  createTimeSchema(),
+			updateTimeKey:  updateTimeSchema(),
+
+			consentPurposeKey: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Purpose of the consent",
+			},
+			consentApplicationIDKey: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Application ID",
+			},
+			consentValidityPeriodKey: {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Specifies the duration in second that the consent remains valid, ranging from 1 day to 2 years",
+			},
+			consentRevokeAfterUseKey: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "If set to true, the consent will be revoked after it is used once",
+			},
+			consentDataPointsKey: {
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Data points is a list of properties related to the Digital twin that the consent is for",
+			},
+		},
+	}
+}
+
+func resourceConsentFlatten(
+	data *schema.ResourceData,
+	resp *configpb.ReadConfigNodeResponse,
+) diag.Diagnostics {
+	var d diag.Diagnostics
+	consentCfg := resp.GetConfigNode().GetConsentConfig()
+	setData(&d, data, consentPurposeKey, consentCfg.GetPurpose())
+	setData(&d, data, consentApplicationIDKey, consentCfg.GetApplicationId())
+	setData(&d, data, consentValidityPeriodKey, consentCfg.GetValidityPeriod())
+	setData(&d, data, consentRevokeAfterUseKey, consentCfg.GetRevokeAfterUse())
+	setData(&d, data, consentDataPointsKey, consentCfg.GetDataPoints())
+	return d
+}
+
+func consentConfigBuilder(data *schema.ResourceData) *configpb.ConsentConfiguration {
+	cfg := &configpb.ConsentConfiguration{
+		Purpose:        data.Get(consentPurposeKey).(string),
+		ApplicationId:  data.Get(consentApplicationIDKey).(string),
+		ValidityPeriod: uint64(data.Get(consentValidityPeriodKey).(int)),
+		RevokeAfterUse: data.Get(consentRevokeAfterUseKey).(bool),
+		DataPoints:     rawArrayToTypedArray[string](data.Get(consentDataPointsKey)),
+	}
+	return cfg
+}
+
+func resourceConsentBuild(
+	_ *diag.Diagnostics,
+	data *schema.ResourceData,
+	_ *ClientContext,
+	builder *config.NodeRequest,
+) {
+	cfg := consentConfigBuilder(data)
+	builder.WithConsentConfig(cfg)
+}
```

Comment:
Nitpick: This seems too superfluous to me. I don't see a need for another function.